odoo/fields.py

```diff
@@ -393,6 +393,12 @@ def _setup_attrs(self, model, name):
             value = self.default
             self.default = value if callable(value) else lambda model: value
 
+        # A related field uses the default value set on the other related
+        # field already. Defining a default value only makes sense with an
+        # inverse method.
+        if self.related and self.default and (self.readonly and not self.inverse):
+            raise ValueError(f"Redundant default on {self}")
```

Comment:
This is the wrong place to put this check.  It should be in `_setup_related_full()`.

Reply:
I would log it as a warning instead.  AFAIK this issue does not have side effects, so it is not unacceptable :wink: